debian/patches/0007-add-bt2390-eetf-and-code-refactor-to-opencl-tonemap.patch

```diff
@@ -910,94 +891,247 @@ Index: FFmpeg/libavfilter/opencl/tonemap.cl
+-    float sig = max(max(rgb.x, max(rgb.y, rgb.z)), 1e-6f);
 +    return s;
 +}
  
--    // Desaturate the color using a coefficient dependent on the signal level
--    if (desat_param > 0.0f) {
--        float luma = get_luma_dst(rgb);
--        float coeff = max(sig - 0.18f, 1e-6f) / max(sig, 1e-6f);
--        coeff = native_powr(coeff, 10.0f / desat_param);
--        rgb = mix(rgb, (float3)luma, (float3)coeff);
--        sig = mix(sig, luma * slope, coeff);
--    }
-+void reshape_dovi_yuv(float3 *yuv,
-+                      __global const vec *src_dovi_params,
-+                      __global const vec *src_dovi_pivots,
-+                      __global const vec4 *src_dovi_coeffs,
-+                      __global const vec4 *src_dovi_mmr)
+-    // Rescale the variables in order to bring it into a representation where
+-    // 1.0 represents the dst_peak. This is because all of the tone mapping
+-    // algorithms are defined in such a way that they map to the range [0.0, 1.0].
+-    if (target_peak > 1.0f) {
+-        sig *= 1.0f / target_peak;
+-        peak *= 1.0f / target_peak;
++vec4 reshape_mmrx4(vec4 sig_i4,
++                   vec4 sig_p4,
++                   vec4 sig_t4,
++                   vec4 coeffsx,
++                   vec4 coeffsy,
++                   vec4 coeffsz,
++                   vec4 coeffsw,
++                   __global const vec4 *dovi_mmr,
++                   uchar dovi_mmr_single,
++                   uchar dovi_min_order,
++                   uchar dovi_max_order)
++{
++    vec4 coeffs = (vec4)(coeffsx.x, coeffsy.x, coeffsz.x, coeffsw.x);
++    vec4 result = (vec4)M_ZERO_VEC;
++    result.x = reshape_mmr((vec3)(sig_i4.x, sig_p4.x, sig_t4.x), coeffs, dovi_mmr,
++                           dovi_mmr_single, dovi_min_order, dovi_max_order);
++    coeffs = (vec4)(coeffsx.y, coeffsy.y, coeffsz.y, coeffsw.y);
++    result.y = reshape_mmr((vec3)(sig_i4.y, sig_p4.y, sig_t4.y), coeffs, dovi_mmr,
++                           dovi_mmr_single, dovi_min_order, dovi_max_order);
++    coeffs = (vec4)(coeffsx.z, coeffsy.z, coeffsz.z, coeffsw.z);
++    result.z = reshape_mmr((vec3)(sig_i4.z, sig_p4.z, sig_t4.z), coeffs, dovi_mmr,
++                           dovi_mmr_single, dovi_min_order, dovi_max_order);
++    coeffs = (vec4)(coeffsx.w, coeffsy.w, coeffsz.w, coeffsw.w);
++    result.w = reshape_mmr((vec3)(sig_i4.w, sig_p4.w, sig_t4.w), coeffs, dovi_mmr,
++                           dovi_mmr_single, dovi_min_order, dovi_max_order);
++    return result;
++}
++
++vec4 reshape_mmr_ptx4(vec4 sig_i4,
++                      vec4 sig_p4,
++                      vec4 sig_t4,
++                      vec4 coeffs,
++                      __global const vec4 *dovi_mmr,
++                      uchar dovi_mmr_single,
++                      uchar dovi_min_order,
++                      uchar dovi_max_order)
 +{
-+    uchar i;
-+    bool t, has_mmr_poly;
++    uint mmr_idx = dovi_mmr_single ? 0 : (uint)coeffs.y;
```

Comment:
Q: Is uint necessary? The maximum allowed value of mmr_idx is only 47.

Reply:
No it is not. I just use uint because I used to.